### DISCUSSION TITLE
Architect Mode

### DISCUSSION THREAD

**Discussion by danividalg** [cline/cline#420] [Feature Requests]
Hi,

Aider has released a new version with something that helps to get better results, it's called "Architect Mode".
https://aider.chat/2024/09/26/architect.html

It could be great if Claude Dev could have something similar.

Thanks a lot :)

**Comment by fabiofurlano** [comment]
Yes it would be a great addition to an already great tool!

**Comment by AJV009** [comment]
NICE, yeah, I came here looking for these.
- Enforce a COT/Plan-like thing, one model for the planning phase and another for the code editing or making changes with those instructions.
- After the plan is generated, we, the users, should have a way to modify and correct it if needed and then hit send for generation and stuff.
- Also, having all config editable in config like continue.dev would be cool.
- I would also love it if the prompts were something I could easily modify as well, like the planning prompts to generation prompts, it could change person to person or how they want to interact with it....

**Reply by danividalg** [reply]
Sounds very well all this features :)

**Comment by Krontx** [comment]
This would be a killer addition, really looking forward to have an architect mode, rate limits in Claude 3.5 Sonnet are a huge drawback to using their model, while DeepSeek has no rate limits at all and cost 30x less. It would be like having the best of all worlds: Pricing , Rate Limits and Reliability.

**Reply by racso-dev** [reply]
If you don't wanna face Anthropic's rate limits just use OpenRouter ;) 5% more expensive, but since they have custom tier agreement with Anthropic no rate limits, or maybe I missed something?

**Comment by PhunkyBob** [comment]
https://github.com/cline/cline/discussions/387

**Reply by danividalg** [reply]
Great, I didn't see your feature request, thanks. 👍😊

**Comment by bendichter** [comment]
Can this be closed with the new Plan/Act feature?

**Comment by phpcommerce** [comment]
This is a fantastic idea, and I believe it could be expanded further with a multi-agent approach. Imagine having a "multi-agent mode" where users interact with specialized AI agents to handle different aspects of the dev process efficiently.

For instance, when starting a new project, you could begin by chatting with a project manager agent. You’d describe your project, its features, and any specific requirements. Based on your input, this agent would suggest a comprehensive plan, including potential tech stacks and architecture. Users could review, modify, or request alternate plans through further conversation until the scope is finalized.

Once the plan is approved, the project manager agent would delegate tasks to developer agents, orchestrating the entire process. Depending on the complexity of the project, multiple developer agents could work collaboratively, each specializing in specific areas, such as backend, frontend, or API integration. Additionally, task delegation could leverage different AI models optimized for specific stages of development: for example, a model like r1 for architectural planning, Sonnet for coding, and another specialized model for testing.

**Reply by dcbartlett** [reply]
Do you mind opening a new Feature Request for the multi-agent idea?  I believe at the current time, Plan/Act suffices to accomplish what this intent of this request was for.

**Comment by dcbartlett** [comment]
Plan/Act is implemented.  This feature allows the end-user to have a conversation with the model to provide a plan of action before implementing.  This help to narrow down exactly what needs to be implemented.
